brickpi3/brickpi3_spi.c

```diff
@@ -138,6 +138,69 @@ int brickpi3_write_u16(struct brickpi3 *bp, enum brickpi3_message msg,
 	return ret;
 }
 
+int brickpi3_write_u8_u16(struct brickpi3 *bp, enum brickpi3_message msg, u8 value1,
+		        u16 value2)
+{
+	int ret;
+
+	mutex_lock(&bp->xfer_lock);
+
+	bp->buf[0] = bp->spi->chip_select;
+	bp->buf[1] = msg;
+	bp->buf[2] = value1;
+	bp->buf[3] = (value2 >> 8) & 0xff;
+	bp->buf[4] = value2 & 0xff;
+	bp->xfer.len = 5;
+
+	ret = spi_sync(bp->spi, &bp->msg);
+
+	mutex_unlock(&bp->xfer_lock);
+
+	return ret;
+}
+
+int brickpi3_set_motor_limits(struct brickpi3 *bp, enum brickpi3_output_port port, u8 duty_cycle_sp,
```

Comment:
I would like to move this function after `brickpi3_i2c_transact()` This way, it is in the same order as it is listed in `brickpi3.h`. And it keeps things a bit more organized since we have the generic helper functions all grouped together here and then all of the other specialized functions later.